packages/core/src/StateMachine.ts

```diff
@@ -653,6 +653,20 @@ export class StateMachine<
       }
     });
 
+    const context = JSON.parse(
+      JSON.stringify(restoredSnapshot.context),
```

Comment:
I don't think we should be using JSON.stringify+JSON.parse dance here. This can legitimately drop values from the `restoredSnapshot.context`. We should write a custom "reviver" here